### PR TITLE
[C-2879] Add validation to single track upload flow

### PR DESCRIPTION
**Pull request by Kyle-Shanks**
### Description

* Adding validation to the single track upload flow using the track metadata schema from sdk.
* Some work needs to be done to update the schema in sdk and then import it here, but it should be an easy hot swap.
* There was a lot of work here to try to bridge the track metadata types between the sdk and common, but open to suggestions on how to clean up some of the checks and data marshalling.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Lots of manual testing

### How will this change be monitored?

N/A

### Feature Flags ###

N/A

### CODE REVIEW

File: packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx [hunk 1]
```diff
@@ -93,19 +105,40 @@ const derivativeWorksValues = [
...
+  .refine((form) => !form[ISWC] || iswcRegex.test(form[ISWC]), {
+    message: messages.iswc.validError,
+    path: [ISWC]
+  })
+
+export type AttributionFormValues = z.input<typeof AttributionFormSchema>
 
 export const AttributionModalForm = () => {
+  // const [{ value: aiUserId }, , { setValue: setAiUserId }] =
+  //   useTrackField<SingleTrackEditValues[typeof AI_USER_ID]>(AI_USER_ID)
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx:139]: 👀

File: packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx [hunk 2]
```diff
@@ -93,19 +105,40 @@ const derivativeWorksValues = [
...
+    [ISWC]: z.optional(z.string().nullable()),
+    [ALLOW_ATTRIBUTION]: z.optional(z.boolean()),
+    [COMMERCIAL_USE]: z.optional(z.boolean()),
+    [DERIVATIVE_WORKS]: z.optional(z.boolean().nullable())
+  })
+  .refine((form) => !form[IS_AI_ATTRIBUTED] || form[AI_USER_ID], {
+    message: messages.aiGenerated.requiredError,
+    path: [AI_USER_ID]
+  })
+  .refine((form) => !form[ISRC] || isrcRegex.test(form[ISRC]), {
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx:126]: sweet

File: packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx [hunk 3]
```diff
@@ -126,6 +159,8 @@ export const AttributionModalForm = () => {
   const initialValues = useMemo(() => {
     const initialValues = {}
     set(initialValues, AI_USER_ID, aiUserId)
+    // @ts-ignore - AI User Id will be a string
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx:162]: should we just cast if we know the type?

File: packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx [hunk 4]
```diff
@@ -185,23 +226,32 @@ export const AttributionModalForm = () => {
...
     >
-      <ModalField
-        title={messages.title}
-        icon={<IconCreativeCommons className={styles.titleIcon} />}
-        preview={preview}
-      >
-        <AttributionModalFields />
-      </ModalField>
+      {() => {
+        return (
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx:233]: could probably remove the brackets and return

File: packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx [hunk 5]
```diff
@@ -234,9 +287,15 @@ const AttributionModalFields = () => {
       >
         <AiAttributionDropdown
           {...aiUserIdField}
-          onSelect={(value: AttributionFormValues[typeof AI_USER_ID]) =>
-            setAiUserId(value)
-          }
+          {...aiUserHelperFields}
+          // @ts-ignore - This is passed down to the dropdown input component that takes error
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx:291]: we could add `type AiAttributionDropdownProps = ComponentPropsWithoutRef<'input'> & ...`

File: packages/web/src/pages/upload-page/forms/SourceFilesModalForm.tsx [hunk 6]
```diff
@@ -24,10 +24,13 @@ import { processFiles } from '../store/utils/processFiles'
 import styles from './SourceFilesModalForm.module.css'
 import { useTrackField } from './utils'
 
+// NOTE: SDK uses camel casing for the fields within download
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/forms/SourceFilesModalForm.tsx:27]: dang that's annoying. Maybe we can align these?

File: packages/web/src/pages/upload-page/forms/SourceFilesModalForm.tsx [hunk 7]
```diff
@@ -107,13 +120,13 @@ export const SourceFilesModalForm = () => {
         icon={<IconSourceFiles className={styles.titleIcon} />}
         preview={preview}
       >
-        <SourceFilesModalFiels />
+        <SourceFilesModalFields />
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/forms/SourceFilesModalForm.tsx:123]: lol whoops

**Review by amendelsohn — APPROVED** [review]
Dang dude. Excellent work! This is a real minefield.

We should chat with @sliptype re: types in sdk. Maybe we can align a bit better and get rid of some of the ts-ignores?

File: packages/web/src/pages/upload-page/fields/TrackMetadataFields.module.css [hunk 8]
```diff
@@ -12,23 +12,28 @@
   flex: 1 auto;
   min-width: 335px;
 }
-.fields > div {
+
+.fields>div {
```
[the remark this answers is not in the document]
Comment (Kyle-Shanks) [reply packages/web/src/pages/upload-page/fields/TrackMetadataFields.module.css:16]: nah this was my linter/formatter doing something weird. will fix

File: packages/web/src/pages/upload-page/components/EditPageNew.tsx [hunk 9]
```diff
@@ -41,27 +53,115 @@ type EditPageProps = {
...
+
+const EditFormValidationSchema = () =>
+  z.object({
+    trackMetadatas: z.array(createTrackMetadataSchema())
+  })
 
 export const EditPageNew = (props: EditPageProps) => {
   const { tracks, setTracks, onContinue } = props
 
+  // @ts-ignore - Slight differences in the sdk vs common track metadata types
```
[the remark this answers is not in the document]
Comment (Kyle-Shanks) [reply packages/web/src/pages/upload-page/components/EditPageNew.tsx:156]: I actually might not need this one, will check

**Pull request merged**